openrarity/cli/opensea.py

```diff
@@ -0,0 +1,106 @@
+import asyncio as aio
+import json
+import logging
+from pathlib import Path
+from typing import Optional
+
+import typer
+
+from openrarity import TokenCollection
+from openrarity.providers import OpenseaApi
+
+from .utils import DEFAULT_COLUMNS, print_rankings
+
+logger = logging.getLogger(__name__)
+app = typer.Typer(name="opensea")
+
+
+@app.command("fetch-assets")
+def fetch_assets(
+    token_ids_file: Path = typer.Argument(
+        None,
+        help="Either a newline delimited text file or json file with an array of token ids.",
+    ),
+    slug: str = typer.Option(..., help="Collection slug to fetch"),
+    start_token_id: Optional[int] = typer.Option(
+        None, help="Start id value for contiguous token_id collections"
+    ),
+    end_token_id: Optional[int] = typer.Option(
+        None, help="End id value for contiguous token_id collections"
+    ),
+    semi_fungible: bool = typer.Option(False),
+    rank: bool = typer.Option(False),
+    write: bool = typer.Option(
+        False,
+        "-w",
+        "--write",
+        help="Boolean flag to write results.",
+    ),
+    dir: Path = typer.Option(
+        ".", "-d", "--dir", help="Directory to write outputs files."
+    ),
+    columns: str = typer.Option(
+        DEFAULT_COLUMNS,
+        "--columns",
+        "-C",
+        help="Columns to print or write to file.",
+    ),
+):
+    if token_ids_file is None:
+        if start_token_id is None or end_token_id is None:
+            raise ValueError(
+                "Either a list of token-ids must be provided or both --start-token-id AND --end-token-id must be set."
```

Comment:
Why? we want All tokens  from collections to be scorred normally

Reply:
This is meant to be a way to provide an apriori range of tokens ids to the Opensea API for making requests. ie: BAYC would be `--start-token-id 0` and `--end-token-id 9999`